src/py/mat3ra/made/tools/calculate.py

```diff
@@ -125,3 +132,44 @@ def calculate_interfacial_energy(
     surface_energy_film = calculate_surface_energy(film_slab, film_bulk, calculator)
     adhesion_energy = calculate_adhesion_energy(interface, substrate_slab, film_slab, calculator)
     return surface_energy_film + surface_energy_substrate - adhesion_energy
+
+
+class InteractionCalculatorParameters(BaseModel):
+    shadowing_radius: float = 2.5
+    interaction_function: Callable = get_sum_of_inverse_distances_squared
+
+
+@decorator_handle_periodic_boundary_conditions(cutoff=0.25)
+def calculate_film_substrate_interaction_metric(
+    material: Material,
+    shadowing_radius: float = 2.5,
+    interaction_function: Callable = get_sum_of_inverse_distances_squared,
```

Comment:
Let's create a class `Calculator` with `interaction_function` as parameter and `get_energy(self, material)` as a class

InterfacialCalculator will have shadowing_radius